src/NavigationHelper/include/WalkingControllers/NavigationHelper/NavigationHelper.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file NavigationHelper.h
+ * @authors Simone Micheletti <simone.micheletti@iit.it>
+ * @copyright 2023 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2023
+ */
+
+#ifndef NAVIGATION_HELPER__HPP
+#define NAVIGATION_HELPER__HPP
+
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Bottle.h>
+#include <atomic>
+#include <thread>
+#include <deque>
+#include <mutex>
+
+namespace WalkingControllers
+{
+    class NavigationHelper
+    {
+    private:
+        yarp::os::BufferedPort<yarp::os::Bottle> m_replanningTriggerPort; /**< Publishes the flag triggering the navigation's global planner. */
+        std::atomic<bool> m_runThreads{false};      /**< Global flag that allows the looping of the threads. */
+        std::deque<bool> m_leftInContact;           /**< Copy of the deques in Module of the left feet contacts status. */
+        std::deque<bool> m_rightInContact;          /**< Copy of the deques in Module of the left feet contacts status. */
+        double m_navigationReplanningDelay;         /**< Delay in seconds of how much to wait before sending the trigger to the navigation stack after exiting double support. */
+        int m_navigationTriggerLoopRate;            /**< Loop rate for the thread computing the navigation trigger*/
+        bool m_publishInfo;                         /**< Flag to whether publish information. */
+        std::thread m_navigationTriggerThread;      /**< Thread for publishing the flag triggering the navigation's global planner. */
+        std::mutex m_updateFeetMutex;    
+        bool m_simulationMode{false};               /**< Flag that syncs the trigger delay with the external clock if in simulation. */  
+
+        const std::string m_portPrefix = "/navigation_helper";
+        
+        /**
+         * Function launched by the looping thread
+         */
+        void computeNavigationTrigger();
+
+    public:
+        NavigationHelper();
+        ~NavigationHelper();
+
+        /**
+         * Close the Navigation Helper Threads and ports
+         * @return true/false in case of success/failure.
+         */
+        bool closeThreads();
```

Comment:
Probably also this can be private.

Reply:
Yes and I would also add in the destructor (the same is valid for `closeHelper`)

Reply:
I have moved it to private and added/fixed some comments/descriptions.
The destructor should never be called inside `closeThreads()`, since the ports could still be open.
In `closeHelper()` I don't see the reason to call the destructor inside a class method, since we are not dealing with special manual memory management. I would avoid that and let the owner handle it.